sparkplug-extras/src/test/scala/springnz/sparkplug/elasticsearch/ElasticSearchTests.scala

```diff
@@ -0,0 +1,315 @@
+package springnz.sparkplug.elasticsearch
+
+import org.apache.spark.rdd.RDD
+import org.scalatest._
+import play.api.libs.json._
+import springnz.sparkplug.core.SparkOperation
+import springnz.sparkplug.testkit.SimpleTestContext
+import springnz.util.{ JsUtil, Logging }
+
+import org.elasticsearch.spark._
+
+import scala.util.{ Try, Success }
+
+trait Fixture {
+  val esServer = new ElasticsearchServer("ESExporterTest")
+
+  val testData = ESExporterTest.testData
+
+  val testDataArray = (Json.parse(testData) match {
+    case JsArray(sequence) ⇒ sequence
+    case _                 ⇒ Seq.empty[JsValue]
+  }) map { jsValue ⇒ Json.stringify(jsValue) }
+}
+
+class ESExporterTest extends WordSpec with Fixture with ShouldMatchers with Logging with BeforeAndAfterAll {
+  override def beforeAll() = {
+    log.info("Elasticsearch server starting...")
+    esServer.start()
+  }
+  override def afterAll() = {
+    log.info("Elasticsearch server stopping...")
+    esServer.stop()
+  }
+
+  "ElasticSearchExporter" should {
+    "Save to ES" in new SimpleTestContext("ESExporterTest") {
+      val operation = SparkOperation { ctx ⇒
+        val rdd = ctx.parallelize(testDataArray)
+        rdd.map { jsonString ⇒
+          val parsedJson: JsValue = Json.parse(jsonString)
+          val unWrappedJson = JsUtil.unwrapFromJs(parsedJson)
+          unWrappedJson.asInstanceOf[Map[String, Any]]
+        }
+      }
+
+      val exporter = new ESExporter[String, Any]("estestindex", "estesttype")
+
+      val mappedResult: SparkOperation[(RDD[Map[String, Any]], Try[Unit])] = for {
+        rdd ← operation
+        exportOp ← exporter.export(rdd)
+      } yield exportOp
+
+      val verify = SparkOperation { ctx ⇒
+        ctx.esJsonRDD("estestindex/estesttype")
+      }
+
+      val verifiedResult = for {
+        (rddOfMaps, tryUnit) ← mappedResult
+        verifiedRdd ← verify
+      } yield (rddOfMaps.collect(), tryUnit, verifiedRdd.collect())
+
+      log.info("Executing Spark process to run and insert...")
+      val result = execute(verifiedResult).get
+
+      result._2 shouldBe a[Success[_]]
+      val mapArray: Array[Map[String, Any]] = result._1
+      mapArray should have length 5
```

Comment:
If the assertion fails, the error printout can be extremely verbose with this check syntax, since It will print out the `toString` of the entire data structure that is being checked. For that reason I always go with `blah.length shouldBe n`